sonar-html-plugin/src/main/java/org/sonar/plugins/html/api/accessibility/ControlGroup.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * SonarSource HTML analyzer :: Sonar Plugin
+ * Copyright (c) 2010-2024 SonarSource SA and Matthijs Galesloot
+ * sonarqube@googlegroups.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sonar.plugins.html.api.accessibility;
+
+import org.sonar.plugins.html.api.HtmlConstants;
+import org.sonar.plugins.html.node.TagNode;
+
+public class ControlGroup {
```

Comment:
Do you foresee using this class in a different context than the rule you implement in this PR?
I just wonder whether we should move it to the rule class.
I am not saying we should, just trying to understand :)

Reply:
My main reasoning was that the rule class was getting very big. I'm not sure though this API makes sense outside of the autocomplete attribute - from my investigation in HTML spec there is no other mention of the concept of "Control group" outside of the autocomplete attribute spec, so it will probably never be used anywhere else...except if we have another rule at some point about this attribute.

Reply:
I see. I don't have a strong opinion on the matter so I let you decide whether you want to leave it there or move it.

Reply:
I'll keep it this way for now if that's OK for you.

Reply:
👍